src/app/components/popup-host/popup-host.component.ts

```diff
@@ -0,0 +1,33 @@
+import {
+    Component,
+    ElementRef,
+    Input,
+    TemplateRef,
+    ViewChild,
+    ViewContainerRef,
+} from '@angular/core';
+
+@Component({
+    selector: 'app-popup-host',
+    templateUrl: './popup-host.component.html',
+    styleUrls: ['./popup-host.component.css'],
+})
+export class PopupHostComponent {
+    @ViewChild('dialog', {read: ElementRef})
+    private readonly dialog: ElementRef | undefined;
+
+    @ViewChild('vc', {read: ViewContainerRef})
+    private readonly vc: ViewContainerRef | undefined;
+
+    @Input()
+    set template(tmpl: TemplateRef<unknown> | null) {
+        if (tmpl) {
+            this.vc?.clear();
+            this.vc?.createEmbeddedView(tmpl);
+            this.dialog?.nativeElement.showModal();
+        } else {
+            this.vc?.clear();
+            this.dialog?.nativeElement.close();
+        }
```

Comment:
Чуть чуть попродаю best practics)

Не рекомендую к использованию конструкцию `if/else`, особенно в комбинации нескольких `else` (`if / else if / else`). Вместо этого рекомендую использовать ранний `return`, он более гибки позволяет делать промежуточны вычисления, и, соответсвенно, писать более чистый код)
```suggestion
        this.vc?.clear();
            
        if (tmpl) {
            this.insertPopupTemplate(templateRef);
            this.dialog?.nativeElement.showModal();
            
            return;
        }
        
        this.dialog?.nativeElement.close();
```